src/fiber/channel.rs

```diff
@@ -654,73 +658,139 @@ where
         state: &mut ChannelActorState,
         commitment_signed: CommitmentSigned,
     ) -> Result<(), ProcessingChannelError> {
+        eprintln!(
+            "handle_commitment_signed_peer_message .... : {}",
+            state.tlc_state.waiting_ack
+        );
         // build commitment tx and verify signature from remote, if passed send ACK for partner
         state.verify_commitment_signed_and_send_ack(commitment_signed, &self.network)?;
-        self.flush_staging_tlc_operations(myself, state).await;
+        let need_commitment_signed = state.tlc_state.update_for_commitment_signed();
+
+        if need_commitment_signed {
+            if !state.tlc_state.waiting_ack {
+                eprintln!("sending commitment_signed after first ack");
+                self.handle_commitment_signed_command(state)?;
+            } else {
+                eprintln!("waiting ack for commitment_signed .............");
+            }
+        }
+
+        // flush remove tlc for received tlcs after replying ack for peer
+        let mut settled_tlcs = vec![];
+        for tlc in state.tlc_state.received_tlcs.tlcs.iter_mut() {
+            if let Some((_removed_at, reason)) = &tlc.removed_at {
+                if matches!(
+                    tlc.status,
+                    TlcStatus::Inbound(InboundTlctatus::RemoveAckConfirmed)
+                ) {
+                    settled_tlcs.push((tlc.tlc_id, reason.clone()));
+                }
+            }
+        }
+        for (tlc_id, reason) in settled_tlcs.iter() {
+            eprintln!("apply received tlc remove: {:?}", tlc_id);
+            let _ = self
+                .apply_remove_tlc_operation(myself, state, *tlc_id, reason.clone())
+                .await;
+        }
+
         Ok(())
     }
 
-    async fn flush_staging_tlc_operations(
+    async fn flush_tlc_operations(
```

Comment:
It is my understanding that `flush_tlc_operations` is called whenever the peer ack'd our `CommitmentSigned` and we want to update the status of those TLCs added/removed in that commitment transaction. The function name `flush_tlc_operations` does not have any implication in this regard. Maybe we can rename this to something like `update_tlc_status_on_ack` to reflect more on why and when to call this function.